packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -186,9 +188,11 @@ export default function MainFeedLayout({
     query = { query: null };
   }
 
-  const [selectedAlgo, setSelectedAlgo] = useState(0);
+  const [selectedAlgo, setSelectedAlgo, loadedAlgo] = usePersistentState(
```

Comment:
This actually sounds like a perfect match for usePersistentContext 🦸‍♀️

Reply:
I actually did use it before. I think there's a bug in it. It doesn't change the value even if I call the second element which should replace the value.